File: Linked_Lists/python/middle.py
from LinkedList import  LinkedList, Node
from random import randint


class StaticVar(object):
    length = 0


def delete_middle(node):
    node.data = node.next.data
    node.next = node.next.next


def get_middle(head: LinkedList, s):
    if head.next is None:
        s.length += 1
        return s.length

    s.length += 1
    index = get_middle(head.next, s) - 1

    if index == s.length // 2:
        delete_middle(head)

    return index


def main():
    linked_list = LinkedList()
    for i in range(5):
        linked_list.append(randint(0, 5))
    print(linked_list)

    s = StaticVar
    get_middle(linked_list.head, s)

    print(linked_list)



if __name__ == '__main__':
    main()

File: Linked_Lists/go/middle.go
package main

import (
  "types"
  "math/rand"
  // "fmt"
)


// Length of linked list
var Length int


func deleteMiddle(node *types.Node) {
  node.Data = node.Next.Data
  node.Next = node.Next.Next
}


func getMiddle(head *types.Node) int {
  if head.Next == nil {
    Length++
    return Length
  }

  Length++
  index := getMiddle(head.Next) - 1

  if index == Length / 2 {
    deleteMiddle(head.Next)
  }
  return index
}


func main() {
  var list types.LinkedList
  var length int = 5

  for i := 0; i < length; i++ {
    list.Append(rand.Intn(23))
  }

  types.PrintLinkedList(list)
  getMiddle(list.Head)
  types.PrintLinkedList(list)
}
